pinakes/main/approval/serializers.py

```diff
@@ -170,6 +172,55 @@ def create(self, validated_data):
         return CreateAction(request, validated_data).process().action
 
 
+class RequestCapabilitiesField(UserCapabilitiesField):
+    """Customized user capabilities for requests"""
+
+    def to_representation(self, value):
+        permissions = super().to_representation(value)
+        return {**permissions, **self._valid_actions(permissions, value)}
+
+    def _valid_actions(self, permissions, request):
+        if not permissions.get("retrieve"):
+            return {}
+
+        valid_actions = {}
+
+        admin = self._is_admin()
+        owner = self._is_owner(request)
+        if admin or owner:
+            valid_actions["cancel"] = RequestCapabilitiesField._can_cancel(
+                request
+            )
+        if admin or not owner:
+            valid_actions["memo"] = True
+            valid_actions["approve"] = RequestCapabilitiesField._can_decide(
+                request
+            )
+            valid_actions["deny"] = valid_actions["approve"]
+        return valid_actions
+
+    def _is_admin(self):
+        return WorkflowPermission().has_permission(
+            self.context["request"], self.context["view"]
+        )
+
+    def _is_owner(self, request):
```

Comment:
It would be nice to have type hints here for disambiguation. A `request` name that is not an http request instance may be confusing.